src/Microsoft.Diagnostics.EventFlow.Outputs.EventHub/EventDataExtensions.cs

```diff
@@ -105,9 +106,15 @@ private static ShoeBoxEventData ConvertToShoeboxMetric(EventData eventData, IRea
                     metricName = metadata["metricName"],
                     time = eventData.Timestamp,
                     timeGrain = null,
-                    dimensions = null,
+                    dimensions = new Dictionary<string, object>()
```

Comment:
dimensions were not filled before, 
`metricRecord.dimensions.Add(nameof(eventData.EventContext), eventData.EventContext);` is called later